torch/_dynamo/symbolic_convert.py

```diff
@@ -277,6 +277,21 @@ def inner(self: "InstructionTranslatorBase", inst: Instruction):
             if truth_fn(value):
                 push and self.push(value)
                 self.jump(inst)
+        elif isinstance(value, UserDefinedObjectVariable):
+            if "__bool__" in value.value.__class__.__dict__:
```

Comment:
This will miss:
1) `__bool__` defined in a base class
2) `__bool__` defined on the instance (we can likely just graph break in this case -- but need to check for it)

You could check `inspect.getattr_static`